### PR TITLE
Fix Pause Mission

### PR DESCRIPTION
**Pull request by AlexisTabin**
- Pausing current mission does not requires mission builder to be ready anymore.
- Mission Items are reset if starting mission fails.

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/ui/maps/MapActivity.kt [hunk 1]
```diff
@@ -277,14 +277,19 @@ class MapActivity : MapViewBaseActivity(), OnMapReadyCallback, MapboxMap.OnMapLo
...
-     * if there are not enough waypoints for a mission
-     * If the drone is on ground -> starts mission
-     * If the drone is flying -> shows return dialog
+     * If the drone is not connected, shows a Toast
+     * If the drone is connected :
+     *      If the drone is already flying, but paused, it restarts it
+     *      If the drone is already flying and doing a mission, it pauses the mission
+     *      If the drone is on ground -
+     *          If there are not enough waypoints for a mission, shows a Toast
+     * If drone is connected, not flying and with a mission builder ready : it starts the mission
```
Comment (Sporarum) [inline app/src/main/java/ch/epfl/sdp/ui/maps/MapActivity.kt:286]: This line can be put like the others, with "if the drone is on ground"

File: app/src/main/java/ch/epfl/sdp/drone/Drone.kt [hunk 2]
```diff
@@ -103,63 +103,47 @@ object Drone {
...
+                        { ToastHandler().showToast(R.string.drone_pause_error, Toast.LENGTH_SHORT) }))
     }
 
     /**
      * Restarts the current Mission
      */
-    private fun restartMission(): Disposable {
-        this.isMissionPausedLiveData.postValue(false)
-        return getConnectedInstance()
+    fun restartMission() {
```
Comment (Ph0tonic) [inline app/src/main/java/ch/epfl/sdp/drone/Drone.kt:138]: Should not it be `resume` the mission` ?

Comment (AlexisTabin) [reply app/src/main/java/ch/epfl/sdp/drone/Drone.kt]: Changed.

File: app/src/main/java/ch/epfl/sdp/drone/Drone.kt [hunk 3]
```diff
@@ -199,15 +206,21 @@ object Drone {
...
                         .subscribe(
-                                { ToastHandler().showToast(R.string.drone_user_success, Toast.LENGTH_SHORT) },
-                                { ToastHandler().showToast(R.string.drone_user_error, Toast.LENGTH_SHORT) }))
+                                {
+                                    this.missionLiveData.value = listOf(DroneUtils.generateMissionItem(returnLocation.latitude, returnLocation.longitude, returnLocation.altitude.toFloat()))
+                                    ToastHandler().showToast(R.string.drone_user_success, Toast.LENGTH_SHORT)
+                                },
+                                {
+                                    ToastHandler().showToast(R.string.drone_user_error, Toast.LENGTH_SHORT)
+                                }
```
Comment (Ph0tonic) [inline app/src/main/java/ch/epfl/sdp/drone/Drone.kt:221]: To be tested.

Comment (AlexisTabin) [reply app/src/main/java/ch/epfl/sdp/drone/Drone.kt:221]: Done

**Pull request merged**